cmd/crank/beta/validate/manager_test.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validate
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	conregv1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/static"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/spf13/afero"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+)
+
+var (
+	configPkg = []byte(`apiVersion: meta.pkg.crossplane.io/v1alpha1
+kind: Configuration
+metadata:
+  name: config-pkg
+spec:
+  dependsOn:
+    - provider: provider-dep-1
+      version: "v1.3.0"
+---
+
+`)
+
+	providerYaml = []byte(`apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-dep-1
+spec:
+  package: provider-dep-1:v1.3.0
+---
+
+`)
+
+	funcYaml = []byte(`apiVersion: pkg.crossplane.io/v1beta1
+kind: Function
+metadata:
+  name: function-dep-1
+spec:
+  package: function-dep-1:v1.3.0
+---
+
+`)
+)
+
+func TestConfigurationTypeSupport(t *testing.T) {
+	confpkg := static.NewLayer(configPkg, types.OCILayer)
+	pd := static.NewLayer(providerYaml, types.OCILayer)
+	fd := static.NewLayer(funcYaml, types.OCILayer)
+
+	fetchMockFunc := func(image string) (*conregv1.Layer, error) {
+		switch image {
+		case "config-pkg:v1.3.0":
+			return &confpkg, nil
+		case "provider-dep-1:v1.3.0":
+			return &pd, nil
+		case "function-dep-1:v1.3.0":
+			return &fd, nil
+		default:
+			return nil, fmt.Errorf("unknown image: %s", image)
+		}
+	}
+
+	type args struct {
+		extensions []*unstructured.Unstructured
+		fetchMock  func(image string) (*conregv1.Layer, error)
+	}
+	type want struct {
+		err   error
+		confs int
+		deps  int
+	}
+	cases := map[string]struct {
+		reason string
+		args   args
+		want   want
+	}{
+		"SuccessfulConfigPkg": {
+			//config-pkg
```

Comment:
nice, this little dependency tree diagram really helps quickly understand what the test case is setting up and testing! 💪